cli/src/lib/zksync-era-state.ts

```diff
@@ -239,63 +231,102 @@ export class ZksyncEraState {
   }
 
   static async fromCalldata(
```

Comment:
This is the most important change. Now in order to create a ZkSyncEraState from an upgrade calldata, instead of parsing the calldata, we execute the upgrade in debug mode using `debug_traceCall`.